File: PySupport/PreCudaParser/TernaryC/Ternary.c
//Simplest C++ Transformation Test File
//initial:: chooser: true  x: 2  y: 6  z: -777
//post::    chooser: true  x: 2  y: 6  z: 2
//initial:: chooser: false  x: 2  y: 6  z: 2
//post::    chooser: false  x: 2  y: 6  z: 6:    chooser: false  x: 2  y: 6  z: 7

#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>

int main()
{
    printf("\nSimplest C++ Transformation Test File");
    int x = 2, y = 6, z = -777;

    bool chooser = true;
    printf("\ninitial:: chooser: %s  x: %d  y: %d  z: %d", chooser ? "true" : "false", x, y, z);
    z = chooser ? x : y;
    printf("\npost::    chooser: %s  x: %d  y: %d  z: %d", chooser ? "true" : "false", x, y, z);
    chooser = false;
    printf("\ninitial:: chooser: %s  x: %d  y: %d  z: %d", chooser ? "true" : "false", x, y, z);
    z = chooser ? x : y;
    printf("\npost::    chooser: %s  x: %d  y: %d  z: %d", chooser ? "true" : "false", x, y, z);
    return 0;
}

File: PySupport/PreCudaParser/TernaryJ/src/apps/Ternary.java
//Java Test ternary IF operator NO side-effects
//
//initial:: chooser:  true  x: 0  y: 0  z: 0
//post::    chooser:  true  x: 1  y: 0  z: 1
//
//initial:: chooser:  false  x: 1  y: 0  z: 1
//post::    chooser:  false  x: 1  y: 1  z: 1

package apps;

class Ternary {

	public static void main(String[] args) {
		int x = 1, y = 5, z = 0;

		System.out.println("\nJava Test ternary IF operator NO side-effects");
		boolean chooser = true;
		System.out.println("\ninitial:: chooser:  " + chooser + "  x: " + x + "  y: " + y + "  z: " + z);
		z = chooser ? x + 1 : y + 1;
		System.out.println("post::    chooser:  " + chooser + "  x: " + x + "  y: " + y + "  z: " + z);
		chooser = false;
		System.out.println("\ninitial:: chooser:  " + chooser + "  x: " + x + "  y: " + y + "  z: " + z);
		z = chooser ? x + 1 : y + 1;
		System.out.println("post::    chooser:  " + chooser + "  x: " + x + "  y: " + y + "  z: " + z);
	}
}